"Error resolving project artifact" unless comes from maven central

I have a tiny maven project that uses the maven repository `https://jcenter.bintray.com/` which is needed to resolve the dependency `org.jfrog.security:jfrog-crypto`.

It compiles and runs without problems, which (I think) proves that the setup is correct.

However, the spdx-maven-plugin fails with the error message mentioned below.

I had expected the spdx-maven-plugin to download `jfrog-crypto-4.9.0.pom` from `https://jcenter.bintray.com/` rather than maven central.

The maven project can be found in the attachment.


```
[INFO] org.spdx.maven.CreateSpdxMojo - Creating SPDX File C:\dvdv\spdx-maven-plugin-minimal-0\target\site\org.example_spdx-maven-plugin-minimal-1.0-SNAPSHOT.spdx.json
Downloading from central: https://repo.maven.apache.org/maven2/org/jfrog/security/jfrog-crypto/4.9.0/jfrog-crypto-4.9.0.pom
[ERROR] org.spdx.maven.CreateSpdxMojo - Maven Project Build Error creating SPDX package for dependency artifact ID jfrog-crypto:Error resolving project artifact: Could not find artifact org.jfrog.security:jfrog-crypto:pom:4.9.0 in central (https://repo.maven.apache.org/maven2) for project org.jfrog.security:jfrog-crypto:jar:4.9.0
[WARNING] org.spdx.maven.CreateSpdxMojo - Error creating SPDX package for dependency artifact ID jfrog-crypto.  A minimal SPDX package will be created.
Downloading from central: https://repo.maven.apache.org/maven2/org/jfrog/common/jfrog-utils/4.9.0/jfrog-utils-4.9.0.pom
[ERROR] org.spdx.maven.CreateSpdxMojo - Maven Project Build Error creating SPDX package for dependency artifact ID jfrog-utils:Error resolving project artifact: Could not find artifact org.jfrog.common:jfrog-utils:pom:4.9.0 in central (https://repo.maven.apache.org/maven2) for project org.jfrog.common:jfrog-utils:jar:4.9.0
[WARNING] org.spdx.maven.CreateSpdxMojo - Error creating SPDX package for dependency artifact ID jfrog-utils.  A minimal SPDX package will be created.
```
[spdx-maven-plugin-minimal-0.zip](https://github.com/spdx/spdx-maven-plugin/files/10990115/spdx-maven-plugin-minimal-0.zip)

--- a/src/main/java/org/spdx/maven/CreateSpdxMojo.java
+++ b/src/main/java/org/spdx/maven/CreateSpdxMojo.java
@@ -47,7 +47,6 @@ import org.spdx.library.model.license.SpdxNoAssertionLicense;
 import java.io.File;
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.net.URLEncoder;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
@@ -584,7 +583,7 @@ public class CreateSpdxMojo extends AbstractMojo
         {
             for ( Artifact dependency : dependencies )
             {
-                retval.addMavenDependency( dependency, mavenProjectBuilder, session );
+                retval.addMavenDependency( dependency, mavenProjectBuilder, session, mavenProject );
             }
         }
         return retval;

--- a/src/main/java/org/spdx/maven/SpdxDependencyInformation.java
+++ b/src/main/java/org/spdx/maven/SpdxDependencyInformation.java
@@ -35,15 +35,18 @@ import java.util.Optional;
 import javax.annotation.Nullable;
 
 import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
 import org.apache.maven.execution.MavenSession;
 import org.apache.maven.model.Contributor;
 import org.apache.maven.model.License;
 import org.apache.maven.model.Model;
 import org.apache.maven.model.Resource;
 import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.project.DefaultProjectBuildingRequest;
 import org.apache.maven.project.MavenProject;
 import org.apache.maven.project.ProjectBuilder;
 import org.apache.maven.project.ProjectBuildingException;
+import org.apache.maven.project.ProjectBuildingRequest;
 import org.apache.maven.project.ProjectBuildingResult;
 import org.apache.maven.shared.model.fileset.FileSet;
 import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
@@ -164,11 +167,12 @@ public class SpdxDependencyInformation
      * @param dependency
      * @param mavenProjectBuilder project builder for the repo containing the POM file
      * @param session Maven session for building the project
+     * @param  mavenProject Maven project
      * @throws LicenseMapperException
      * @throws InvalidSPDXAnalysisException 
      */
     public void addMavenDependency( Artifact dependency, ProjectBuilder mavenProjectBuilder, 
-                                    MavenSession session ) throws LicenseMapperException, InvalidSPDXAnalysisException
+                                    MavenSession session, MavenProject mavenProject ) throws LicenseMapperException, InvalidSPDXAnalysisException
     {
         String scope = dependency.getScope();
         RelationshipType relType = scopeToRelationshipType( scope, dependency.isOptional() );
@@ -177,7 +181,7 @@ public class SpdxDependencyInformation
             log.warn(
                     "Could not determine the SPDX relationship type for dependency artifact ID " + dependency.getArtifactId() + " scope " + scope );
         }
-        SpdxElement dependencyPackage = createSpdxPackage( dependency, mavenProjectBuilder, session );
+        SpdxElement dependencyPackage = createSpdxPackage( dependency, mavenProjectBuilder, session, mavenProject );
         if ( relType.toString().endsWith( "_OF" ))
         {
             if ( dependencyPackage instanceof SpdxPackage)
@@ -236,12 +240,14 @@ public class SpdxDependencyInformation
      * @param artifact Maven dependency artifact
      * @param mavenProjectBuilder project builder for the repo containing the POM file
      * @param session Maven session for building the project
+     * @param mavenProject Maven project
      * @return SPDX Package build from the MavenProject metadata
      * @throws LicenseMapperException
      * @throws InvalidSPDXAnalysisException 
      */
     private SpdxElement createSpdxPackage( Artifact artifact, 
-                                           ProjectBuilder mavenProjectBuilder, MavenSession session ) throws LicenseMapperException, InvalidSPDXAnalysisException
+                                           ProjectBuilder mavenProjectBuilder, MavenSession session,
+                                           MavenProject mavenProject ) throws LicenseMapperException, InvalidSPDXAnalysisException
     {
         log.debug( "Creating SPDX package for artifact " + artifact.getArtifactId() );
         if ( artifact.getFile() == null )
@@ -306,7 +312,16 @@ public class SpdxDependencyInformation
         }
         try
         {
-            ProjectBuildingResult build = mavenProjectBuilder.build(artifact, session.getProjectBuildingRequest());
+            ProjectBuildingRequest request = new DefaultProjectBuildingRequest( session.getProjectBuildingRequest() );
+            request.setProject( mavenProject );
+            request.setRemoteRepositories( mavenProject.getRemoteArtifactRepositories() );
+            for (ArtifactRepository ar:request.getRemoteRepositories()) {
+                log.debug( "request Remote repository ID: " + ar.getId() );
+            }
+            for (ArtifactRepository ar:mavenProject.getRemoteArtifactRepositories()) {
+                log.debug( "Project Remote repository ID: " + ar.getId() );
+            }
+            ProjectBuildingResult build = mavenProjectBuilder.build( artifact, request );
             MavenProject depProject = build.getProject();
             log.debug(
                       "Dependency " + artifact.getArtifactId() + "Collecting information from project metadata for " + depProject.getArtifactId() );

